src/prefect/engine.py

```diff
@@ -2395,6 +2400,48 @@ async def _run_task_hooks(task: Task, task_run: TaskRun, state: State) -> None:
                 logger.info(f"Hook {hook_name!r} finished running successfully")
 
 
+async def _check_task_failure_retriable(
+    task: Task, task_run: TaskRun, state: State
+) -> bool:
+    """Run the `retry_condition_fn` callable for a task, making sure to catch and log any errors
+    that occur. Defaults to True if no `retry_condition_fn` is set.
+    """
+    # By default, a task is retriable if it is in a `Failed` state
+    if task.retry_condition_fn is None:
+        return True
+
+    logger = task_run_logger(task_run)
+    try:
+        logger.debug(
+            f"Running `retry_condition_fn` check {task.retry_condition_fn!r} for task"
+            f" {task.name!r}"
+        )
+        if is_async_fn(task.retry_condition_fn):
+            return await task.retry_condition_fn(
+                task=task, task_run=task_run, state=state
+            )
```

Comment:
Can we check if `task.retry_condition_fn` is callable first so we can surface a nicer error?

Reply:
Callable check added to log the type and traceback.

Reply:
I think we should validate this during task creation (in `tasks.py`).